backend/app/user/supabase/supabase_user_service.py

```diff
@@ -66,36 +68,31 @@ def __upsert_into_database(
             .execute()
         )
 
-    def __retrieve_user_id(self, authentication_token: AuthToken) -> UserId:
-        """
-        Return a user ID that corresponds to an authentication_token.
-
-        If no such user ID is found, return an anonymous user ID.
-        """
-
-        try:
-            user_response = self.__auth_service.get_user(authentication_token)
-        except AuthException as exception:
-            raise UserServiceException from exception
-
-        if user_response.user_id:
-            return user_response.user_id
-
-        try:
-            return self.__auth_service.sign_in_anonymously().user_id
-        except AuthException as exception:
-            raise UserServiceException from exception
-
-    def create_user_from_id(self, user_id: UserId) -> User:
+    def create_user_from_id(self, *, user_id: UserId) -> User:
         """Return a new User, with an ID that matches user_id."""
 
         return User(id=user_id, _service=self)
 
-    def create_user_from_token(self, authentication_token: AuthToken) -> User:
+    def create_user_from_token(self, *, authentication_token: AuthToken) -> User:
         """
         Retrieve a user ID from __auth_service, and return a new User with a matching ID.
         """
-        return self.create_user_from_id(self.__retrieve_user_id(authentication_token))
+
+        try:
+            return self.create_user_from_id(
+                user_id=self.__auth_service.get_user(
+                    authentication_token=authentication_token
+                ).user_id
+            )
+        except AuthException:
+            try:
```

Comment:
Discussed on meet.

Should probably introspect the exception to make sure it's recoverable. There will be exceptions that don't arise from the user not being signed in.

Reply:
I have added `UserNotFoundAuthException` to handle the cases where user info isn't in the auth API.